src/_guides/language/concurrency/index.md

```diff
@@ -315,104 +321,136 @@ to implement isolates.
-[`ReceivePort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/ReceivePort-class.html
-[`SendPort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/SendPort-class.html
+#### Running an existing method in a new isolate
 
 Here’s the code for the main isolate:
 
 <?code-excerpt "lib/simple_worker_isolate.dart (main)"?>
 ```dart
 void main() async {
   // Read some data.
-  final jsonData = await _parseInBackground();
+  final jsonData = await Isolate.run(_readAndParseJson);
 
   // Use that data
   print('Number of JSON keys: ${jsonData.length}');
 }
+```
+
+The spawned isolate executes the following code:
 
-// Spawns an isolate and waits for the first message
-Future<Map<String, dynamic>> _parseInBackground() async {
-  final p = ReceivePort();
-  await Isolate.spawn(_readAndParseJson, p.sendPort);
-  return await p.first as Map<String, dynamic>;
+<?code-excerpt "lib/simple_worker_isolate.dart (spawned)"?>
+```dart
+Future<Map<String, dynamic>> _readAndParseJson() async {
+  final fileData = await File(filename).readAsString();
+  final jsonData = jsonDecode(fileData) as Map<String, dynamic>;
+  return jsonData;
 }
 ```
 
-The `_parseInBackground()` function contains the code that
-_spawns_ (creates and starts) the isolate for the background worker,
-and then returns the result:
+1. `Isolate.run()` spawns an isolate for the background worker,
+   while `main()` waits for the result.
 
-1. Before spawning the isolate, the code creates a `ReceivePort`,
-   which enables the worker isolate
-   to send messages to the main isolate.
+2. The new isolate executes the argument passed to `run()`:
+   the function `_readAndParseJson()`.
 
-2. Next is the call to `Isolate.spawn()`,
-   which creates and starts the isolate for the background worker.
-   The first argument to `Isolate.spawn()` is the function that
-   the worker isolate executes: `_readAndParseJson`.
-   The second argument is the `SendPort`
-   that the worker isolate can use to send messages to the main isolate.
-   The code doesn't _create_ a `SendPort`;
-   it uses the `sendPort` property of the `ReceivePort`.
+3. The `return` statement ends the computation,
+   shutting down the worker isolate and
+   sending `jsonData` back to the main isolate as the result.
 
-3. Once the isolate is spawned, the main isolate waits for the result.
-   Because the `ReceivePort` class implements `Stream`,
-   the [`first`][] property is an easy way to get
-   the single message that the worker isolate sends.
+In this example, `_readAndParseJson()` is an existing,
+asynchronous function that could just as easily
+run directly in the main isolate.
+But using `Isolate.run()` enables concurrency,
+so its computations are completely abstracted by the worker isolate
+and can complete without blocking the main isolate.
 
-[`first`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/Stream/first.html
+An added benefit is that the result of
```

Comment:
Being asynchronous is not what people would normally call a "benefit" :)

I'd say rather just not value-load it, and say something like:

> Because `Isolate.run()` allows other code to run while the computation
> executed in another isolate, its result is always returned as a `Future`.
> To the original isolate, it doesn't matter whether the computation running in the 
> other isolate is synchronous or asynchronous, because it's being run concurrently
> either way.